### DISCUSSION TITLE
safetyProfile design spec

### DISCUSSION THREAD

**Discussion by jwildfire** [SafetyGraphics/safetyProfile#3] [General]
safetyProfile is a shiny module that serves as a patient profile for clinical trial data. Requirements/features include: 
- Compatible with safetyGraphics Shiny app
    - Expects standard safetyGraphics `data()` and `settings()` reactives as inputs. 
    - Updates as expected with mapping and filter updates in safetyGraphics app. 
    - Implemented as a custom tab (like the filter module) instead of a standard chart 
       - this will require a PR to the core safetyGraphics library at some point
- Updates dynamically based on available data
   - The module should recognize the standard data domains supported in safetyGraphics/safetyCharts and provide useful charts for key domains. 
   - domain-specific graphics should shown/hidden based upon which domains are provided in `data()`

**Comment by jwildfire** [comment]
Notes from @xni7: 
- Add clinician
- Consider data requirements
- Two use cases: 
    - Select each participant
    - Link from other charts

**Comment by samussiah** [comment]
Need an [approach](https://github.com/SafetyGraphics/safetyProfile/issues/4) to study timing to display each domain by study day when not available.
